prow/prstatus/prstatus.go

```diff
@@ -323,7 +323,11 @@ func (da *DashboardAgent) QueryPullRequests(ctx context.Context, ghc githubClien
 func (da *DashboardAgent) GetHeadContexts(ghc githubClient, pr PullRequest) ([]Context, error) {
 	org := string(pr.Repository.Owner.Login)
 	repo := string(pr.Repository.Name)
-	combined, err := ghc.GetCombinedStatus(org, repo, string(pr.HeadRefOID))
+	ref := string(pr.HeadRefOID)
+	if org == "" || repo == "" || ref == "" {
+		return nil, fmt.Errorf("combined status asked for empty pull request: %v", pr)
```

Comment:
This seems OK but is symptomatic of bad caller behavior so I think it would be good to also dig into why someone is sending empty `pr` objects.

Reply:
Do you mean check with GitHub, or check earlier in the code - perhaps in `HandlePrStatus` or `QueryPullRequests`?

Reply:
Earlier in the code -- unless we think we're getting empty objects from GitHub, which I don't think is the case.

Reply:
But @stevekuznetsov I think this is exactly what @cjwagner meant here: https://github.com/kubernetes/test-infra/issues/12902#issuecomment-499200246

Reply:
Err but that is upstream of this callsite, right? What calls `GetHeadContexts` with an empty `pr`? I guess it does make sense to be defensive _here_, maybe, but I would like it more to catch that where we initialize the `nil` value for the PR in the first place.

Reply:
Alright, now I understand what you meant :)